### PR TITLE
Make sure we don't fetch chunks that don't span the start point

### PR DESCRIPTION
**Pull request by mildbyte**

### CODE REVIEW

File: arctic/tickstore/tickstore.py [hunk 1]
```diff
@@ -176,17 +176,28 @@ def _mongo_date_range_query(self, symbol, date_range):
...
+                        break
             except StopIteration:
                 pass
+            
         if first_dt:
             start_range['$gte'] = first_dt
+        else:
+            # If all chunks start inside of the range, make sure
+            # we don't fetch everything
+            start_range['$gte'] = start
```
Comment (jamesblackburn) [inline arctic/tickstore/tickstore.py]: Is this edge-case tested too?

Comment (mildbyte) [inline arctic/tickstore/tickstore.py]: It is now :)

**Comment by bmoscon** [comment]
looks good to me.

**Comment by jamesblackburn** [comment]
LGTM @mildbyte 

Don't forget to update the Changelog :)

**Pull request merged**